nomos-da/network/subnetworks-assignations/src/lib.rs

```diff
@@ -26,4 +29,40 @@ pub trait MembershipHandler {
     fn members(&self) -> HashSet<Self::Id>;
 
     fn last_subnetwork_id(&self) -> Self::NetworkId;
+
+    fn get_address(&self, peer_id: &PeerId) -> Option<Multiaddr>;
+}
+
+use std::sync::Arc;
+
+impl<T> MembershipHandler for Arc<T>
+where
+    T: MembershipHandler,
+{
+    type NetworkId = T::NetworkId;
+    type Id = T::Id;
+
+    fn membership(&self, id: &Self::Id) -> HashSet<Self::NetworkId> {
+        (**self).membership(id)
```

Comment:
I think in this cases it would be clearer to use `as_ref`
```suggestion
        self.as_ref().membership(id)
```

Or fully qualified syntax:
```suggestion
        Membership::membership(self.as_ref(), id)
```

Reply:
Good catch, fixed! Ty